Sources/FluentSQL/SQLDatabase+Model.swift

```diff
@@ -0,0 +1,48 @@
+import SQLKit
+
+extension SQLQueryFetcher {
+    public func first<Model>(decoding model: Model.Type) -> EventLoopFuture<Model?> 
+        where Model: FluentKit.Model
+    {
+        self.all(decoding: Model.self).map { $0.first }
+    }
+
+    public func all<Model>(decoding model: Model.Type) -> EventLoopFuture<[Model]> 
+        where Model: FluentKit.Model
+    {
+        self.all().flatMapThrowing { rows in 
+            try rows.map { row in
+                let model = Model()
+                try model.output(from: SQLDatabaseOutput(sql: row))
+                return model
```

Comment:
This would be useful on its own for row decoding. I already see a case where I would want to decode the row as a Model, in addition to these high level helpers.

So something like this:
```swift
extension SQLRow {
    public func decode<Model>(model _: Model.Type) throws -> Model where Model: FluentKit.Model {
        let model = Model()
        try model.output(from: SQLDatabaseOutput(sql: self))
        return model
    }
}
```

Reply:
Added!